code/js/modules/BaseController.js

```diff
@@ -1,30 +1,39 @@
-;(function() {
+(function() {
   "use strict";
 
+  require("../modules/utils.js");
   var sk_log = require("../modules/SKLog.js");
 
   function BaseController(options) {
     this.siteName = options.siteName || null;
 
-    this.selectors = {
-      //** Properties **//
-      playPause: (options.playPause || null),
-      play: (options.play || null),
-      pause: (options.pause || null),
-      playNext: (options.playNext || null),
-      playPrev: (options.playPrev || null),
-      mute: (options.mute || null),
-      like: (options.like || null),
-      dislike: (options.dislike || null),
-      iframe: (options.iframe || null),
-
-      //** States **//
-      playState: (options.playState || null),
-
+    this.selectors = {};
+    var selectorsKeys = [
+      'playPause', 'play', 'pause',
+      'playNext', 'playPrev', 'isPlaying',
+      'mute', 'like', 'dislike', 'iframe',
       //** Song Info **//
-      song: (options.song || null),
-      artist: (options.artist || null)
-    };
+      'song', 'artist'
+    ];
+    selectorsKeys.forEach((key) => this.selectors[key] = null);
+
+    // Assign passed options to this and this.selectors
+    Object.keys(options).forEach((key) => {
```

Comment:
This project already has `lodash` as a dependency. es6 is cool and all, but this function could more or less be 2 lines of lodash operations - and inherently more reliable as there's no custom code to test